dbms/src/Functions/tests/gtest_strings_elt.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <ext/range.h>
+
+namespace DB
+{
+namespace tests
+{
+class TestFunctionElt : public DB::tests::FunctionTest
+{
+};
+
+#define ASSERT_ELT(expected, t1, t2, ...) \
+    ASSERT_COLUMN_EQ(expected, executeFunction("elt", t1, t2, __VA_ARGS__))
+
+TEST_F(TestFunctionElt, BoundaryIdx)
+try
+{
+    const ColumnWithTypeAndName dummy_col0 = createColumn<Nullable<String>>({"abc", "123", "", {}, ""});
+    const ColumnWithTypeAndName dummy_col1 = createColumn<Nullable<String>>({"def", "123", {}, "", ""});
+
+    static constexpr size_t ndummy_col = 5;
```

Comment:
Maybe we can assign the variable with a dynamic method, for example `ndummy_col = dummy_col0.size()`, instead of hard code.

Reply:
sure